api/organizations.go

```diff
@@ -234,3 +239,179 @@ func (a *API) updateOrganizationHandler(w http.ResponseWriter, r *http.Request)
+	if _, err := a.db.IsMemberOf(invite.Email, org.Address, db.AdminRole); err == nil {
+		ErrDuplicateConflict.With("user is already admin of organization").Write(w)
+		return
+	}
+	// create new invitation
+	inviteCode := internal.RandomHex(VerificationCodeLength)
+	if err := a.db.CreateInvitation(&db.OrganizationInvite{
+		InvitationCode:      inviteCode,
+		OrganizationAddress: org.Address,
+		NewUserEmail:        invite.Email,
+		Role:                db.UserRole(invite.Role),
+		CurrentUserID:       user.ID,
+		Expiration:          time.Now().Add(InvitationExpiration),
+	}); err != nil {
+		ErrGenericInternalServerError.Withf("could not create invitation: %v", err).Write(w)
+		return
+	}
+	// send the invitation email
+	ctx, cancel := context.WithTimeout(r.Context(), time.Second*10)
+	defer cancel()
+	// send the verification code via email if the mail service is available
+	if a.mail != nil {
+		if err := a.mail.SendNotification(ctx, &notifications.Notification{
+			ToName:    fmt.Sprintf("%s %s", user.FirstName, user.LastName),
+			ToAddress: invite.Email,
+			Subject:   InvitationEmailSubject,
+			Body:      fmt.Sprintf(InvitationTextBody, org.Name, inviteCode),
+		}); err != nil {
+			ErrGenericInternalServerError.Withf("could not send verification code: %v", err).Write(w)
+			return
+		}
+	}
+	httpWriteOK(w)
+}
+
+// acceptOrganizationMemberInvitationHandler handles the request to accept an
+// invitation to an organization. It checks if the invitation is valid and not
+// expired, and if the user is not already a member of the organization. If the
+// user does not exist, it creates a new user with the provided information.
+// If the user already exists and is verified, it adds the organization to the
+// user.
+func (a *API) acceptOrganizationMemberInvitationHandler(w http.ResponseWriter, r *http.Request) {
+	// get the organization info from the request context
+	org, _, ok := a.organizationFromRequest(r)
+	if !ok {
+		ErrNoOrganizationProvided.Write(w)
+		return
+	}
+	// get new member info from the request body
+	invitationReq := &AcceptOrganizationInvitation{}
+	if err := json.NewDecoder(r.Body).Decode(invitationReq); err != nil {
+		ErrMalformedBody.Write(w)
+		return
+	}
+	// get the invitation from the database
+	invitation, err := a.db.Invitation(invitationReq.Code)
+	if err != nil {
+		ErrUnauthorized.Withf("could not get invitation: %v", err).Write(w)
+		return
+	}
+	// check if the organization is correct
+	if invitation.OrganizationAddress != org.Address {
+		ErrUnauthorized.Withf("invitation is not for this organization").Write(w)
+		return
+	}
+	// create a helper function to remove the invitation from the database in
+	// case of error or expiration
+	removeInvitation := func() {
+		if err := a.db.DeleteInvitation(invitationReq.Code); err != nil {
+			log.Warnf("could not delete invitation: %v", err)
+		}
+	}
+	// check if the invitation is expired
+	if invitation.Expiration.Before(time.Now()) {
+		go removeInvitation()
```

Comment:
With the goroutine, the invitation remove process occurs in background, if it fails, the error will be logged but not blocks the user request. The expiration time is checked before, so if the code is expired it will not be validate any way.

If the deletion is deferred, it could not returns an error but it will block the request.

So, in any case, as the error is no handled and just logged, there is not difference between defer it or execute it in background, only that it will not block the user request.